frontend/src/components/Plan/Plan.tsx

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { useEffect, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom';
+import { icon } from '@fortawesome/fontawesome-svg-core/import.macro'
+import styles from './Plan.module.css'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { PlanDescription } from './PlanDescription'
+import { Memberslist } from './MembersList'
+import { fetchPlanData } from './hook/fetchPlanData'
+import { loadingStatus } from '../../utils/loadingStatus'
+import { PlanIdeaList } from './PlanIdeaList'
+
+export interface Member {
+  leader: string
+  friends: FriendList[]
+}
+
+export interface FriendList {
+  firstName: string
+  lastName: string
+  id: string
+}
+
+export const Plan = (): JSX.Element => {
+  const { id } = useParams()
+
+  const [data, status] = fetchPlanData(id)
+
+  if (status === loadingStatus.LOADING || status === loadingStatus.UNLOADED) {
+    return (
+      <div>
+        <p>loading plan</p>
+      </div>
+    )
+  } else if (status === loadingStatus.FAILED) {
+    return (
+      <div>
+        <p>couldn't fetch plan</p>
+      </div>
+    )
+  }
+  const specialPerson = (data.specialPerson)
+  const description = (data.description)
+  const specialDate = (data.specialDate)
+  const members: Member = (data.members)
```

Comment:
neat trick you can use called restructuring:

```
const {specialPerson, description, specialDate, members} = data
```

Reply:
yes this is better!